File: MATTHEW/leetcode/AIcoding/4.GPT-4o/go_solutions/2364-count-number-of-bad-pairs.go
func countBadPairs(nums []int) int64 {
    n := len(nums)
    goodPairs := 0
    for i := 0; i < n; i++ {
        goodPairs += i - nums[i]
    }
    totalPairs := n * (n - 1) / 2
    return int64(totalPairs - goodPairs)
}

File: leetcode/AIcoding/2.Gemini-2.0/cpp_solutions/2364-count-number-of-bad-pairs.cpp
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
public:
    long long countBadPairs(vector<int>& nums) {
        int n = nums.size();
        long long totalPairs = (long long)n * (n - 1) / 2;
        unordered_map<int, int> count;
        long long goodPairs = 0;

        for (int i = 0; i < n; i++) {
            int diff = nums[i] - i;
            if (count.find(diff) != count.end()) {
                goodPairs += count[diff];
            }
            count[diff]++;
        }

        return totalPairs - goodPairs;
    }
};